File: csharp-selenium-nunit-reqnroll/SauceDemo.UiTests/Configuration/ParallelSettings.cs
namespace SauceDemo.UiTests.Configuration;

/// <summary>
/// Single source of truth for how many scenarios run concurrently. Must stay in sync with
/// <see cref="Users.UserCatalog.PoolUsers"/> (one distinct login user per parallel worker) and is
/// consumed by the compile-time <c>LevelOfParallelismAttribute</c> in AssemblyInfo.cs, which requires
/// a constant.
/// </summary>
public static class ParallelSettings
{
    public const int MaxParallelWorkers = 3;
}

File: python-selenium-pytest-bdd/src/saucedemo_uitests/config/parallel_settings.py
"""Single source of truth for how many scenarios run concurrently.

Must stay in sync with `saucedemo_uitests.users.user_catalog.POOL_USERS` (one distinct login user per
parallel worker) and with the `-n` value passed to `pytest-xdist` in scripts/CI - unlike the two other
frameworks in this repo, there is no compile-time-constant attribute to thread this value into (NUnit's
`[assembly: LevelOfParallelism(...)]`, Cucumber.js's `cucumber.js` config), so keeping this in sync is
entirely a matter of convention, enforced only by `assigned_user.py`'s own runtime assertion.
"""

MAX_PARALLEL_WORKERS = 3
